### PR TITLE
Dev piotrek checkstyle and random placement fixes

### PR DESCRIPTION
**Pull request by sabath91**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java [hunk 1]
```diff
@@ -16,47 +23,62 @@
...
   }
 
-  private List<Integer> offerShipAsIndices(final int shipLength, final int startIndex) {
+  private List<Integer> offerShipAsIndices(final int shipLength) {
+    int startIndex = this.generateStartIndex();
     List<Integer> shipAsIndices;
     boolean verticalOrientation = this.isVerticalOrientation();
     if (verticalOrientation) {
-      shipAsIndices = this.offerIndices(shipLength, startIndex, 10);
+      shipAsIndices = this.offerIndices(shipLength, startIndex, BOARD_WIDTH);
```
Comment (M-Aarsman) [inline server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java]: BOARD_WIDTH is in class so no need to send to class method

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java [hunk 2]
```diff
@@ -76,7 +98,14 @@ private void markAreaAsOccupied(final List<Integer> offeredShip) {
         .forEach(i -> gameBoard.put(i, FieldState.OCCUPIED));
   }
 
-  private boolean isWithinBoard(final List<Integer> offeredShip) {
+  private boolean isWithinBoardAndSingleRow(final List<Integer> offeredShip) {
+    int rowNumberOfFirstShipIndex = offeredShip.get(0) / BOARD_WIDTH;
+    return offeredShip
+        .stream()
+        .allMatch(i -> i < 100 && (i / BOARD_WIDTH == rowNumberOfFirstShipIndex));
```
Comment (M-Aarsman) [inline server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java]: divide to filter and any match

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java [hunk 3]
```diff
@@ -76,7 +98,14 @@ private void markAreaAsOccupied(final List<Integer> offeredShip) {
...
+  }
+
+  private boolean isWithinBoardVertical(final List<Integer> offeredShip) {
     return offeredShip
         .stream()
         .allMatch(i -> i > 0 && i < 100);
```
Comment (M-Aarsman) [inline server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java]: same here

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java (hunk 1 above)
Comment (sabath91) [reply server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java]: done

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java (hunk 2 above)
Comment (sabath91) [reply server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java]: It works in a different way.

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java (hunk 3 above)
Comment (sabath91) [reply server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetGenerator.java]: It works in a different way.

**Pull request merged**